azure-pipelines/end-to-end-tests-dir/asset-caching.ps1

```diff
@@ -69,159 +80,252 @@ if (-not ($actual -match $expected)) {
+"note: If you are using a proxy, please ensure your proxy settings are correct\.",
+"Possible causes are:",
+"1\. You are actually using an HTTP proxy, but setting HTTPS_PROXY variable to ``https//address:port``\.",
+"This is not correct, because ``https://`` prefix claims the proxy is an HTTPS proxy, while your proxy \(v2ray, shadowsocksr, etc\.\.\.\) is an HTTP proxy\.",
+"Try setting ``http://address:port`` to both HTTP_PROXY and HTTPS_PROXY instead\."
+"2\. If you are using Windows, vcpkg will automatically use your Windows IE Proxy Settings set by your proxy software\. See: https://github\.com/microsoft/vcpkg-tool/pull/77",
+"The value set by your proxy might be wrong, or have same ``https://`` prefix issue\.",
+"3\. Your proxy's remote server is our of service\.",
+"If you've tried directly download the link, and believe this is not a temporary download server failure, please submit an issue at https://github\.com/Microsoft/vcpkg/issues",
+"to report this upstream download server failure\."
+) -join "`n"
+
 $actual = Run-VcpkgAndCaptureOutput -TestArgs ($commonArgs + @("x-download", "$downloadsRoot/example3.html", "--sha512", "d06b93c883f8126a04589937a884032df031b05518eed9d433efb6447834df2596aebd500d69b8283e5702d988ed49655ae654c1683c7a4ae58bfa6b92f2b73a", "--url", "https://localhost:1234/foobar.html", "--x-asset-sources=x-azurl,file://$AssetCache,,readwrite"))
-if (-not ($actual.Contains("Asset cache miss; downloading from https://localhost:1234/foobar.html") -and
-          $actual.Contains("Downloading example3.html") -and
-          $actual.Contains("error: file://$AssetCache") -and
-          $actual.Contains("curl failed to download with exit code 37"))) {
+Throw-IfNotFailed
+if (-not ($actual -match $expected)) {
     throw "Failure: azurl (yes), x-block-origin (no), asset-cache (miss), download (fail)"
 }
 
 # azurl (yes), x-block-origin (no), asset-cache (hit), download (n/a)
 # Expected: Download success message, asset cache named, nothing about x-block-origin
 Refresh-TestRoot
-$actual = $actual -replace "`r`n", "`n"
-Run-VcpkgAndCaptureOutput -TestArgs ($commonArgs + @("x-download", "$downloadsRoot/example3.html", "--sha512", "d06b93c883f8126a04589937a884032df031b05518eed9d433efb6447834df2596aebd500d69b8283e5702d988ed49655ae654c1683c7a4ae58bfa6b92f2b73a", "--url", "https://example.com", "--x-asset-sources=x-azurl,file://$AssetCache,,readwrite"))
+Run-Vcpkg -TestArgs ($commonArgs + @("x-download", "$downloadsRoot/example3.html", "--sha512", "d06b93c883f8126a04589937a884032df031b05518eed9d433efb6447834df2596aebd500d69b8283e5702d988ed49655ae654c1683c7a4ae58bfa6b92f2b73a", "--url", "https://example.com", "--x-asset-sources=x-azurl,file://$AssetCache,,readwrite"))
+Throw-IfFailed
+Remove-Item "$downloadsRoot/example3.html"
+$expected = @(
+"Trying to download example3\.html using asset cache file://$assetCacheRegex/[0-9a-z]+",
+"Download successful! Asset cache hit, did not try authoritative source https://example\.com"
```

Comment:
Same as previous comment. I will add that in the opposite , Asset cache miss', scenario we simply say "Asset cache miss; trying authoritative source".

Reply:
Are you saying you want to remove `Download successful!`?

Reply:
I think we should avoid saying "Download" when we are talking about pulling from the asset cache. I know it's technically the same thing, but I suspect people will conceptually think of them differently. 

That said, I'm speculating and these are all nitpicks so I don't feel strongly either way.

Reply:
If we don't say 'download' what should we say instead? I think it is important to be clear that we are going to touch the network each time before we do that.